tests/unit_tests/test_sky/utils/test_cli_utils.py

```diff
@@ -377,3 +376,83 @@ def test_get_resources_kubernetes():
     # Test K8s TPU resources
     resources_str = status_utils._get_resources(mock_record_k8s_tpu)
     assert resources_str == '1x(gpus=tpu-v4-8:1, cpus=8, mem=32, ...)'
+
+
+def test_get_resources_fractional_cpu():
+    from sky.utils import resources_utils
+
+    # Test fractional CPU (0.5)
+    mock_resources_k8s_fractional = Resources(infra='k8s/my-cluster-ctx',
+                                              cpus='0.5',
+                                              memory=4,
+                                              instance_type='0.5CPU--4GB')
+
+    simple, full = resources_utils.format_resource(
+        mock_resources_k8s_fractional)
+
+    assert 'cpus=0.5' in simple and 'cpus=0.5' in full
+    assert not re.search(r'\bcpus=0(?![.\d])', simple)
+    assert not re.search(r'\bcpus=0(?![.\d])', full)
+
+    # Test fractional CPU with plus sign (0.5+)
+    mock_resources_k8s_fractional_plus = Resources(infra='k8s/my-cluster-ctx',
+                                                   cpus='0.5+',
+                                                   memory=4,
+                                                   instance_type='0.5CPU--4GB')
+    simple_plus, full_plus = resources_utils.format_resource(
+        mock_resources_k8s_fractional_plus)
+    assert 'cpus=0.5' in simple_plus and 'cpus=0.5' in full_plus
+
+    # Test decimal CPU (4.5)
+    mock_resources_k8s_decimal = Resources(infra='k8s/my-cluster-ctx',
+                                           cpus='4.5',
+                                           memory=8,
+                                           instance_type='4.5CPU--8GB')
+    simple_decimal, full_decimal = resources_utils.format_resource(
+        mock_resources_k8s_decimal)
+    assert 'cpus=4.5' in simple_decimal and 'cpus=4.5' in full_decimal
+
+    # Test integer CPU (4)
+    mock_resources_k8s_int = Resources(infra='k8s/my-cluster-ctx',
+                                       cpus='4',
+                                       memory=8,
+                                       instance_type='4CPU--8GB')
+    simple_int, full_int = resources_utils.format_resource(
+        mock_resources_k8s_int)
+    assert 'cpus=4' in simple_int and 'cpus=4' in full_int
+
+
+def test_get_resources_fractional_memory():
```

Comment:
What about we combine the cases into the CPU ones, just build and check both CPU and memory in the same case?